core/src/ids.rs

```diff
@@ -224,41 +230,66 @@ impl ReservationId {
     }
 }
 
+/// Trait representing the extra info needed for program id generation.
+/// This is usually a MessageId, but it can be a handle in the form of a u32 or u64.
+pub trait ProgramNonce {
+    /// Turn the nonce into a vector of bytes to be used in the program id generation.
+    fn as_nonce(&self) -> Vec<u8>;
```

Comment:
Not sure how anything like that would help, since its always a dynamic amount of bytes, so it cannot be a statically allocated anything. u32 gives 4 bytes, u64 gives 8 and MessageId gives way more

Reply:
Actually the array len can be generic.
Anyway, implement this and you can get rid of this trait https://github.com/gear-tech/gear/pull/3010#pullrequestreview-1584247512

Reply:
I'm wondering if I even need the trait, instead just opting to always using a u64, since I don't seem to use the messageid, instead I use the outgoing length.